src/containers/widget/index.js

```diff
@@ -45,12 +47,23 @@ class Widget extends Component {
   }
 
   componentDidMount() {
-    this.setDefaults();
+    return fetch(`${this.props.apiBaseUrl}/v1/configuration`,
+        { headers: {"Api-Token": this.props.apiToken} })
+      .then(response => {
+        if(response.ok) {
+          response.json().then(config => {
+            this.widgetConfig = config;
+            this.setDefaults(config);
+          });
+        } else {
+          this.setDefaults();
+        }
+      });
   }
 
   setDefaults = () => {
     this.setState({
-      ...getInitialState(this.props),
+      ...getInitialState(this.props, this.widgetConfig),
```

Comment:
you can take the config as an argument and then pass it to getInitialState function right?

Reply:
yes, but not from the other setDefaults() calls.

Reply:
yeah, in that case on getInitialState, you can do something like:

```
function getInitialState(props = {}) {
  return Object.assign(Widget.defaultProps, props);
}
```

When you don't pass any argument then the default value will be an empty object.

Reply:
that is what is done here: https://github.com/bumi/area402-widget/pull/85/files#diff-128edf9b24e12b671c98d03163f11c4380faf0f88888c2c76a909cc786e167f2R22

still the question is what to do in the `componentWillUnmount` case? there we also call `setDefaults`

Reply:
in `componentWillUnmount`, you can call setDefaults with `Widget.defaultProps`

Reply:
that would remove all  settings when the modal is opened a second time.